src/main/java/org/jitsi/videobridge/adaptivity/MediaStreamTrackFactory.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright @ 2015 Atlassian Pty Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.videobridge.adaptivity;
+
+import net.java.sip.communicator.impl.protocol.jabber.extensions.colibri.*;
+import net.java.sip.communicator.impl.protocol.jabber.extensions.jingle.*;
+import org.jitsi.impl.neomedia.*;
+
+import java.util.*;
+
+/**
+ * A factory of {@link MediaStreamTrackImpl}s from jingle signaling.
+ *
+ * @author George Politis
+ */
+public class MediaStreamTrackFactory
+{
+    /**
+     * Creates {@link MediaStreamTrackImpl}s from signaling params.
+     *
+     * @param mediaStreamTrackReceiver the {@link MediaStreamTrackReceiver} that
+     * will receive the created {@link MediaStreamTrackImpl}s.
+     * @param sources The {@link List} of {@link SourcePacketExtension} that
+     * describes the list of jingle sources.
+     * @param sourceGroups The {@link List} of
+     * {@link SourceGroupPacketExtension} that describes the list of jingle
+     * source groups.
+     * @return the {@link MediaStreamTrackReceiver} that are described in the
+     * jingle sources and source groups.
+     */
+    public static MediaStreamTrackImpl[] createMediaStreamTracks(
+        MediaStreamTrackReceiver mediaStreamTrackReceiver,
+        List<SourcePacketExtension> sources,
+        List<SourceGroupPacketExtension> sourceGroups)
+    {
+        boolean hasSources = sources != null && !sources.isEmpty();
+        boolean hasGroups = sourceGroups != null && !sourceGroups.isEmpty();
+
+        if (!hasSources && !hasGroups)
+        {
+            return null;
+        }
+
+        List<MediaStreamTrackImpl> tracks = new ArrayList<>();
+        List<Long> grouped = new ArrayList<>();
+
+        if (hasGroups)
+        {
+            List<SourceGroupPacketExtension> simGroups = new ArrayList<>();
+            Map<Long, Long> rtxPairs = new TreeMap<>();
+
+            for (SourceGroupPacketExtension sg : sourceGroups)
+            {
+                List<SourcePacketExtension> groupSources = sg.getSources();
+                if (groupSources == null || groupSources.isEmpty())
+                {
+                    continue;
+                }
+
+                if ("sim".equalsIgnoreCase(sg.getSemantics())
```

Comment:
We have the SEMANTICS_SIMULCAST and SEMANTICS_FID constants in SourceGroupPacketExtension